extension/src/lttb.rs

```diff
@@ -95,6 +96,130 @@ pub fn lttb_final_inner(
     }
 }
 
+#[pg_extern(immutable, parallel_safe, schema = "toolkit_experimental")]
+pub fn gp_lttb_trans(
+    state: Internal,
+    time: crate::raw::TimestampTz,
+    val: Option<f64>,
+    gap: crate::raw::Interval,
+    resolution: i32,
+    fcinfo: pg_sys::FunctionCallInfo,
+) -> Option<Internal> {
+    let state = unsafe { state.to_inner() };
+    let needs_interval = state.is_none();
+
+    // Don't love this code, but need to compute gap_val if needed before time is moved
```

Comment:
I can see why they were reluctant to define copy on datum, since a datum may actually be a pointer pretending to be a usize.  However it might be nice to implement a utility to give us a shallow copy, that we can use for things like timestamptz (which is legitimately a usize).